dart/dynamics/Joint.h

```diff
@@ -119,6 +126,14 @@ class Joint
   /// Get number of generalized coordinates
   DEPRECATED(4.1) virtual size_t getDof() const = 0;
 
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual DegreeOfFreedom* getDof(size_t _index) = 0;
+
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual const DegreeOfFreedom* getDof(size_t _index) const = 0;
```

Comment:
I was debating whether or not to do that. I decided not to because the Skeleton is responsible for DOF name management (because each DOF in the skeleton must have a unique name), so accessing DOFs by name should be done through the skeleton.

Having the joints also keep track of DOF names seems redundant. And if we gave the joint class a function that uses the skeleton to look up DOFs it could be strange/confusing if someone asks a joint for a DOF that belongs to another joint.

In my mind, the thing that makes the most sense is asking the Skeleton for access to a DOF if you want to access by name, and you should only go through the Joint to access a DOF if you want to access it by its index within that Joint.

Let me know if that doesn't make sense.